telegram/ext/_chatjoinrequesthandler.py

```diff
@@ -43,17 +44,41 @@ async def callback(update: Update, context: CallbackContext)
 
             The return value of the callback is usually ignored except for the special case of
             :class:`telegram.ext.ConversationHandler`.
+        chat_id(:obj:`int` | Collection[:obj:`int`], optional): Filters requests to allow only
+            those which are from a specified chat ID.
+
+            .. versionadded:: 20.0
```

Comment:
This currently only allows chat IDs. The issue also requested usernames. Can you implement that as well?
The issue said in one argument, I would be fine with two as well, do not really mind.